sorting/Bogo_Sort.cpp

```diff
@@ -31,7 +42,8 @@ void show_array(const std::vector <int> &arr)
     std::cout << '\n';
 }
 //This function shuffles the array randomly
-std::vector <int> shuffle(std::vector <int> arr)
+template<typename T, size_t N>
```

Comment:
```suggestion

/**
 * Function to shuffle the elements of an array.
 * @tparam T typename of the array
 * @tparam N length of array
 * @param arr array to shuffle
 * @returns new array with elements shuffled from a given array
 */
template<typename T, size_t N>
```
Also, isn't `array.size()` always equal to `N` and the variable can be used instead of a function call.
It would be nice to mention that this function has `arr` as *pass-by-value` and hence will always create a new copy and not modify the original array.